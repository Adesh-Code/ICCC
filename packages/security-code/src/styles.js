const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
