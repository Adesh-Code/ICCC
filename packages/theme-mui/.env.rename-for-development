# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE