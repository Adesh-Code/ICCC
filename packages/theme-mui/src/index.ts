// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
