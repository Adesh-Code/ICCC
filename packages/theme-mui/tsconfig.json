{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
