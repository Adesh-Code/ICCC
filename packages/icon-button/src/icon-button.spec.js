import React from "react";
import { render, screen } from "@testing-library/react";
import IconButton from "./icon-button";
import SvgIcon from "@mui/material/SvgIcon";
import SyncIcon from "@iccc/icon/lib/build/icons/sync";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = (props) => {
  const utils = render(
    <ThemeProvider theme={light}>
      <IconButton {...props}>
        <SvgIcon titleAccess="test">
          <SyncIcon />
        </SvgIcon>
      </IconButton>
    </ThemeProvider>
  );
  return utils;
};

describe("IconButton component", () => {
  beforeEach(() => {
    // eslint-disable-next-line no-console
    console.error = jest.fn((error) => {
      throw new Error(error);
    });
  });

  it("renders a button", () => {
    const { container } = arrange();
    const button = screen.getByRole("button");
    expect(button).toHaveClass("DhigIconButton--variant--contained");
    expect(container).toMatchSnapshot();
  });

  it("accepts classnames", () => {
    arrange({
      className: "foo"
    });
    const button = screen.getByRole("button");
    expect(button).toHaveClass("foo");
  });

  it("renders a stretched button", () => {
    arrange({
      fullWidth: true
    });
    const button = screen.getByRole("button");
    expect(button).toHaveClass("DhigIconButton--fullWidth");
  });

  it("renders an outlined button", () => {
    arrange({
      variant: "outlined"
    });
    const button = screen.getByRole("button");
    expect(button).toHaveClass("DhigIconButton--variant--outlined");
  });
});
