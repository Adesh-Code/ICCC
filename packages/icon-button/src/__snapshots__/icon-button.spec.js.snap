// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component renders a button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium icccIconButton--variant--contained emotion-0"
    tabindex="0"
    type="button"
  >
    <svg
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-1"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="actions/sync"
        >
          <path
            d="M8.3534 14.0279L3.55809 12.4698L2 17.2651M3.88056 12.928C4.9899 19.0317 13.5516 23.139 18.6759 15.5263M15.179 9.97208L19.9743 11.5302L21.5324 6.73486M19.6518 11.072C18.5425 4.96825 9.98077 0.86096 4.85646 8.4737"
            id="stroke"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </g>
      </svg>
      <title>
        test
      </title>
    </svg>
  </button>
</div>
`;
